x-pack/legacy/plugins/graph/public/services/save.tsx

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { showSaveModal } from 'ui/saved_objects/show_saved_object_save_modal';
+import { SavedGraphWorkspace } from '../types/persistence';
+import { GraphSaveModal, OnSaveGraphProps } from '../components/graph_save_modal';
+import { GraphSavePolicy } from '../types/config';
+
+export function save({
+  savePolicy,
+  hasData,
+  workspace,
+  saveWorkspace,
+}: {
+  savePolicy: GraphSavePolicy;
+  hasData: boolean;
+  workspace: SavedGraphWorkspace;
+  saveWorkspace: (
+    saveOptions: {
+      confirmOverwrite: boolean;
+      isTitleDuplicateConfirmed: boolean;
+      onTitleDuplicate: () => void;
+    },
+    dataConsent: boolean
+  ) => Promise<{ id?: string } | { error: string }>;
+}) {
+  const currentTitle = workspace.title;
+  const currentDescription = workspace.title;
+  const onSave = ({
+    newTitle,
+    newDescription,
+    newCopyOnSave,
+    isTitleDuplicateConfirmed,
+    onTitleDuplicate,
+    dataConsent,
+  }: OnSaveGraphProps) => {
+    workspace.title = newTitle;
+    workspace.description = newDescription;
+    workspace.copyOnSave = newCopyOnSave;
+    const saveOptions = {
+      confirmOverwrite: false,
+      isTitleDuplicateConfirmed,
+      onTitleDuplicate,
+    };
+    return saveWorkspace(saveOptions, dataConsent).then(response => {
+      // If the save wasn't successful, put the original values back.
+      if (!('id' in response) || !Boolean(response.id)) {
```

Comment:
While testing I encountered  a problem here in Browsers !== Chrome, response in Firefox and Safari is undefined
<img width="684" alt="Bildschirmfoto 2019-09-02 um 06 42 31" src="https://user-images.githubusercontent.com/463851/64091189-58f3e480-cd4f-11e9-8f6d-4888cc4ea1e7.png">

Reply:
Good catch! Shame on me for not testing this - fixed by passing through the response in all cases.

Reply:
👍